Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -1131,6 +1137,41 @@ private void HandlePositionAssigned(OrderEvent fill)
             _algorithm.OnAssignmentOrderEvent(fill);
         }
 
+        /// <summary>
+        /// Option expiration event is received and new order events are generated
+        /// </summary>
+        private void HandleOptionPositionExpired(OptionPositionExpiredEventArgs e)
+        {
+            if (_algorithm.Securities.TryGetValue(e.Symbol, out var security))
+            {
+                Log.Trace("BrokerageTransactionHandler.HandleOptionPositionExpired(): clearing position for expired option holding: " +
+                    $"Symbol: {e.Symbol.Value}, Quantity: {security.Holdings.Quantity}");
+
+                var quantity = -security.Holdings.Quantity;
+
+                // generate new order and ticket for the expired option
+                var order = new OptionExerciseOrder(e.Symbol, quantity, DateTime.UtcNow)
+                {
+                    Id = _algorithm.Transactions.GetIncrementOrderId()
+                };
+
+                var ticket = order.ToOrderTicket(_algorithm.Transactions);
+
+                AddOpenOrder(order, ticket);
+
+                Interlocked.Increment(ref _totalOrderCount);
+
+                // generate the order events reusing the option exercise model
+                var option = (Option)security;
+                var orderEvents = option.OptionExerciseModel.OptionExercise(option, order).ToList();
```

Comment:
Pretty cool!
I think there is room for removing the BacktestingBrokerage (+ AlgorithmManager) weirdness/special handling to start using this new system,

nit - `ToList()` isn't required I think?